theano/tensor/subtensor.py

```diff
@@ -1972,6 +1978,10 @@ def c_code(self, node, name, input_names, output_names, sub):
         {
             Py_XDECREF(%(out)s);
             %(out)s = %(copy_of_x)s;
+            if (!%(out)s) {
+                PyErr_SetString(PyExc_MemoryError, "Could not allocate copy of input.");
```

Comment:
I am reasonably certain that the CPU code already sets an exception and 100% certain that the GPU code does.